### PR TITLE
About their role forms - Add same organisation page

### PR DESCRIPTION
**Pull request by vassyz**
### Context

Employer can select if the person they are referring works in the same organisation.
An error message will be shown when submitting without selecting a value.

https://trello.com/c/c46KiqWB/987-about-their-role-forms-add-same-organisation-page
https://teacher-misconduct.herokuapp.com/report/teacher-role/school

### Changes proposed in this pull request

Adds the following page:
Screenshot 2022-11-21 at 17 32 38

- Update the routes
- Generate a migration for adding the same organisation field to the referrals table.
- Add the form, including validation
- Link the form to the job title page

### Checklist

- [x] Attach to Trello card
- [] Rebased main
- [] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by vassyz** [comment]
[About their role forms: Add same organisation page](https://trello.com/c/c46KiqWB/987-about-their-role-forms-add-same-organisation-page)

File: db/schema.rb [hunk 1]
```diff
@@ -75,45 +74,56 @@
     t.index ["referral_id"], name: "index_organisations_on_referral_id"
   end
 
+  create_table "referral_evidences", force: :cascade do |t|
```
Comment (steventux) [inline db/schema.rb:77]: This doesn't seem related to this PR, more likely an artifact of reviewing my PR?

**Review by steventux — COMMENTED** [review]
Looks good but schema file appears to have more changes than introduced by the migration in this PR.

File: db/schema.rb (hunk 1 above)
Comment (vassyz) [reply db/schema.rb:77]: Not sure how I missed that. I'll fix it.

**Review by steventux — APPROVED** [review]
:rocket:
